models/activities/notification.go

```diff
@@ -136,18 +136,12 @@ func GetNotifications(ctx context.Context, options *FindNotificationOptions) (nl
 }
 
 // CountNotifications count all notifications that fit to the given options and ignore pagination.
-func CountNotifications(opts *FindNotificationOptions) (int64, error) {
-	return db.GetEngine(db.DefaultContext).Where(opts.ToCond()).Count(&Notification{})
+func CountNotifications(ctx context.Context, opts *FindNotificationOptions) (int64, error) {
+	return db.GetEngine(ctx).Where(opts.ToCond()).Count(&Notification{})
 }
 
 // CreateRepoTransferNotification creates  notification for the user a repository was transferred to
-func CreateRepoTransferNotification(doer, newOwner *user_model.User, repo *repo_model.Repository) error {
-	ctx, committer, err := db.TxContext()
-	if err != nil {
-		return err
-	}
-	defer committer.Close()
-
```

Comment:
I'm not sure that it's right to get rid of the transaction here.

Reply:
Added the transaction again. Maybe we should specify where (at which level) you should start transactions. Currently I think everything deeper than at router or service level is bad because it shows too much logic in a lower layer.

Reply:
#21756 maybe allow us to have a flexible function. We can create a transaction dynamically when we can't find a transaction.